templates/prometheus.go

```diff
@@ -45,5 +53,26 @@ var PrometheusTemplate = promv1.Prometheus{
 			}},
 		},
 		Resources: utils.GetResourceRequirements("prometheus"),
+		Web:       PrometheusWebSpecTemplate.DeepCopy(),
+	},
+}
+
+var locSecretRef = corev1.LocalObjectReference{
+	Name: PrometheusServiceSecretName,
+}
+
+var PrometheusWebSpecTemplate = &promv1.WebSpec{
+	TLSConfig: &promv1.WebTLSConfig{
+		KeySecret: corev1.SecretKeySelector{
+			LocalObjectReference: locSecretRef,
+			Key:                  "tls.key",
+		},
+		Cert: promv1.SecretOrConfigMap{
+			Secret: &corev1.SecretKeySelector{
+				LocalObjectReference: locSecretRef,
+				Key:                  "tls.crt",
+			},
+		},
+		ClientAuthType: fmt.Sprint(tls.VerifyClientCertIfGiven),
```

Comment:
No need for a second template, this is just a sub-struct under the prom spec, please embed it there (currently you create a copy for no reason)

Reply:
Agreed